### ISSUE TITLE
AVT (Label): Input and Textarea don't have an `id` attribute

### ISSUE THREAD

**Issue by stanislavgeorgiev** [IBM/carbon-components-angular#1609]
## Detailed description

> Describe in detail the issue you're having.

Using `ibm-label` together with an `input` or `textarea` element doesn't generate a `id` attribute matching with the label.
```
      <ibm-label>
        My Label
        <textarea ibmTextArea></textarea>
      </ibm-label>
```

Generates the following html code:

```
<ibm-label _ngcontent-cho-c2204="" class="bx--form-item">
   <label class="bx--label" for="ibm-label-0" ng-reflect-ng-class="[object Object]"> My Label </label>
   <div class="bx--text-area__wrapper">
      <textarea _ngcontent-cho-c2204="" ibmtextarea="" class="bx--text-area"></textarea>
   </div>
</ibm-label>
```

> Is this a feature request (new component, new icon), a bug, or a general issue?

Bug

> Is this issue related to a specific component?

`ibm-label`

> What did you expect to happen? What happened instead? What would you like to see changed?

The `textarea` should've had an `id="ibm-label-0"`

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?
```
    "@angular/core": "10.1.3",
    "carbon-components": "10.20.0",
    "carbon-components-angular": "4.17.0",
```
> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

IBM AD

**Comment by GSSCDylan** [comment]
div element within ibm-label has the same issue.

**Comment by cal-smith** [comment]
@GSSCDylan The `div` issue is somewhat expected - we query explicitly for `input` and `textarea` ... We can also query for `[inputElement]` or something similar if that would help?

**Comment by stanislavgeorgiev** [comment]
Our AVT team is interested in how do you see this issue fixed in time?

**Comment by cal-smith** [comment]
I don't have a specific timeline at the moment ... Is there a date you (or your AVT team) need this by?

**Comment by GSSCDylan** [comment]
Hi @cal-smith ,if the issue can be fixed by Nov 25th will be great, this issue impacts lots of panel for our product.

**Comment by stanislavgeorgiev** [comment]
Pretty much the same for us too. Our end of release is in the beginning of December, which makes Nov 27th the last day before the code freeze.

**Comment by zvonimirfras** [comment]
Looking at the [story](https://angular.carbondesignsystem.com/?path=/story/components-input--textarea) and inspecting the code it appears that textarea _does_ have the `id` that matches the `for` attribute value.

The selector we use in label.component.ts is `input,textarea,div` so div should do the same...

What am I misunderstanding here? 🤔

**Comment by cal-smith** [comment]
@zvonimirfras there's some context missing in the issue I suppose ... Our storybook uses Angular 7, but @stanislavgeorgiev is using Angular 9/10 ... Since we've set `static: false` in the "wrapper" `ViewChild` the element isn't available in `ngAfterContentInit` which then prevents the `id` from being applied. This works fine in Angular 7 since it ignores the `static` option. I think the fix should be as simple as changing it to apply the `id` in `ngAfterViewInit`.

**Comment by stanislavgeorgiev** [comment]
We're on Angular 10 now

**Comment by zvonimirfras** [comment]
Ok, I'll take a look at it.
